shared/rakefiles/xk_util.rake

```diff
@@ -0,0 +1,36 @@
+# This task rotates KMS key version and associated secrets file for target args[:encryption_key].
+#
+# If args[:secret] is set, it also rotates the target secret.
+# New value for the secret can be set via env var TF_VAR_secret_name,
+# otherwise new value will be generated automatically.
+# Old secret value will be set to TF_VAR_secret_name_rotated until rotation is finished.
+#
+# Arbitrary command to execute after rotation can be set with :cmd argument.
+task :rotate_secrets, [:encryption_key, :secret, :cmd] => [:configure_serviceaccount] do |taskname, args|
+  if args[:encryption_key].nil? || args[:encryption_key].size == 0
+    puts "  ERROR: Argument :encryption_key not present!"
+    raise
+  end
+
+  @secrets = Secrets.collect_secrets()
+
+  if @secrets[args[:encryption_key]].nil?
+    puts "  ERROR: Encryption key '#{args[:encryption_key]}' does not exist!"
+    raise
+  elsif args[:secret] and ENV["TF_VAR_#{args[:secret]}"].nil?
+    puts "  ERROR: Secret '#{args[:secret]}' does not exist!"
+    raise
+  end
+
+  Secrets.set_secrets(@secrets)
+
+  if args[:secret]
+    ENV["TF_VAR_#{args[:secret]}_rotated"] = ENV["TF_VAR_#{args[:secret]}"]
+    ENV["TF_VAR_#{args[:secret]}"] = ""
+  end
+
+  Secrets.rotate_key(args[:encryption_key])
```

Comment:
I agree that KMS key rotation is a potentially dangerous operation, mostly because there is no way to wrap it up into a single "transaction" that can be reverted automatically in case of `:rotate_secrets_key` task gets terminated / loses internet connectivity in the middle of the rotation.

However, there is also no way to irreversibly lose any sensitive data if situation from previous point happens. Disabled KMS key version can be re-enabled again in case we stuck with the secrets file encrypted with it. And secrets GS bucket is versioned in case we need to recover previous version of secrets file.

I experimented with all sorts of interruptions to `:rotate_secrets_key`, `:rotate_secret`  tasks and I was able to recover from any of failures with minimum effort.